### ISSUE TITLE
rustix-0.38.23+ broken

### ISSUE THREAD

**Issue by SUPERCILEX** [bytecodealliance/rustix#936]
Try upgrading in this project for example: https://github.com/SUPERCILEX/forkfs.

```
error[E0425]: cannot find value `CLOCK_REALTIME_ALARM` in module `c`
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/clockid.rs:70:33
    |
70  |     RealtimeAlarm = bitcast!(c::CLOCK_REALTIME_ALARM),
    |                                 ^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/backend/linux_raw/c.rs:267:1
    |
267 | pub(crate) const CLOCK_REALTIME_COARSE: c_int = linux_raw_sys::general::CLOCK_REALTIME_COARSE as _;
    | --------------------------------------------------------------------------------------------------- similarly named constant `CLOCK_REALTIME_COARSE` defined here
    |
note: found an item that was configured out
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/backend/linux_raw/c.rs:279:18
    |
279 | pub(crate) const CLOCK_REALTIME_ALARM: c_int = linux_raw_sys::general::CLOCK_REALTIME_ALARM as _;
    |                  ^^^^^^^^^^^^^^^^^^^^
help: a constant with a similar name exists
    |
70  |     RealtimeAlarm = bitcast!(c::CLOCK_REALTIME_COARSE),
    |                                 ~~~~~~~~~~~~~~~~~~~~~
help: consider importing this constant
    |
1   + use linux_raw_sys::general::CLOCK_REALTIME_ALARM;
    |
help: if you import `CLOCK_REALTIME_ALARM`, refer to it directly
    |
70  -     RealtimeAlarm = bitcast!(c::CLOCK_REALTIME_ALARM),
70  +     RealtimeAlarm = bitcast!(CLOCK_REALTIME_ALARM),
    |

error[E0425]: cannot find value `CLOCK_TAI` in module `c`
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/clockid.rs:75:23
    |
75  |     Tai = bitcast!(c::CLOCK_TAI),
    |                       ^^^^^^^^^ not found in `c`
    |
note: found an item that was configured out
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/backend/linux_raw/c.rs:277:18
    |
277 | pub(crate) const CLOCK_TAI: c_int = linux_raw_sys::general::CLOCK_TAI as _;
    |                  ^^^^^^^^^
help: consider importing this constant
    |
1   + use linux_raw_sys::general::CLOCK_TAI;
    |
help: if you import `CLOCK_TAI`, refer to it directly
    |
75  -     Tai = bitcast!(c::CLOCK_TAI),
75  +     Tai = bitcast!(CLOCK_TAI),
    |

error[E0425]: cannot find value `CLOCK_BOOTTIME` in module `c`
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/clockid.rs:87:28
    |
87  |     Boottime = bitcast!(c::CLOCK_BOOTTIME),
    |                            ^^^^^^^^^^^^^^
    |
   ::: /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/backend/linux_raw/c.rs:263:1
    |
263 | pub(crate) const CLOCK_REALTIME: c_int = linux_raw_sys::general::CLOCK_REALTIME as _;
    | ------------------------------------------------------------------------------------- similarly named constant `CLOCK_REALTIME` defined here
    |
note: found an item that was configured out
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/backend/linux_raw/c.rs:273:18
    |
273 | pub(crate) const CLOCK_BOOTTIME: c_int = linux_raw_sys::general::CLOCK_BOOTTIME as _;
    |                  ^^^^^^^^^^^^^^
help: a constant with a similar name exists
    |
87  |     Boottime = bitcast!(c::CLOCK_REALTIME),
    |                            ~~~~~~~~~~~~~~
help: consider importing this constant
    |
1   + use linux_raw_sys::general::CLOCK_BOOTTIME;
    |
help: if you import `CLOCK_BOOTTIME`, refer to it directly
    |
87  -     Boottime = bitcast!(c::CLOCK_BOOTTIME),
87  +     Boottime = bitcast!(CLOCK_BOOTTIME),
    |

error[E0425]: cannot find value `CLOCK_BOOTTIME_ALARM` in module `c`
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/clockid.rs:92:33
    |
92  |     BoottimeAlarm = bitcast!(c::CLOCK_BOOTTIME_ALARM),
    |                                 ^^^^^^^^^^^^^^^^^^^^ not found in `c`
    |
note: found an item that was configured out
   --> /home/asaveau/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.24/src/backend/linux_raw/c.rs:275:18
    |
275 | pub(crate) const CLOCK_BOOTTIME_ALARM: c_int = linux_raw_sys::general::CLOCK_BOOTTIME_ALARM as _;
    |                  ^^^^^^^^^^^^^^^^^^^^
help: consider importing this constant
    |
1   + use linux_raw_sys::general::CLOCK_BOOTTIME_ALARM;
    |
help: if you import `CLOCK_BOOTTIME_ALARM`, refer to it directly
    |
92  -     BoottimeAlarm = bitcast!(c::CLOCK_BOOTTIME_ALARM),
92  +     BoottimeAlarm = bitcast!(CLOCK_BOOTTIME_ALARM),
    |
```

**Comment by SUPERCILEX** [comment]
```
rustc 1.76.0-nightly (283170175 2023-11-17)
binary: rustc
commit-hash: 2831701757eb7b3105eda26a306c2f3a97e2664b
commit-date: 2023-11-17
host: x86_64-unknown-linux-gnu
release: 1.76.0-nightly
LLVM version: 17.0.5
```

**Comment by SUPERCILEX** [comment]
A bisection shows that this broke in 0.38.23 and occurs on stable rust too.

**Comment by sunfishcode** [comment]
This is fixed in #927, which I've now released in rustix 0.38.25.

**Comment by SUPERCILEX** [comment]
Confirmed fixed, thanks!
